planet/order_request.py

```diff
@@ -199,6 +199,10 @@ def delivery(archive_type: Optional[str] = None,
     if archive_type:
         archive_type = specs.validate_archive_type(archive_type)
 
+        # for missing archive options set single_archive to false
+        if archive_filename is None and single_archive is False:
+            single_archive = True
```

Comment:
@JSee98 @kevinlacaille I have a different perspective on the problem. I don't think we want to change the single/multi flag, and thus the *shape* of the delivered artifacts, based on other inputs. Instead I think we want to make sure that the archive_filename defaults to some value if it is not specified. After a close read of https://developers.planet.com/apis/orders/delivery/#zipping-results, I believe we want something like this:

```python
if archive_type:
    archive_filename = archive_filename or "{{name}}_{{order_id}}.zip"
```

Reply:
So just to confirm @sgillies, your suggestion is if both the variables are undefined i.e. archive_filename and single_archive we should give a default value to archive_filename?
If you and @kevinlacaille agree on this suggested path I can make the required changes.

Reply:
@JSee98 we should provide a default for `archive_filename` whenever archive_type evaluates to True, no matter the value of `single_archive`. @kevinlacaille asked the API team and found that the need to specify an archive filename is a known Orders API bug and that this Python project is the right place to work around it.

Reply:
Sure guys, will make the updates and raise the same.

Reply:
@sgillies I've made the updates